src/pineko/check.py

```diff
@@ -168,40 +169,37 @@ def contains_ren(grid, max_as, max_al):
              max as order
     max_al : int
              max al order
+    sv_type : Scale
+        kind of scale_variation to be checked (either REN or FACT)
+
     Returns
     -------
-    bool
-        is ren scale-variation available for as
-    bool
-        is ren scale-variation available for al
+    : AvailableAtMax
+        result of the check
+    : int
+        effective max_as in the grid
     """
-    order_array = np.array([order.as_tuple() for order in grid.orders()])
-    order_mask = pineappl.grid.Order.create_mask(grid.orders(), max_as, max_al)
-    order_list = order_array[order_mask]
-    as_orders = []
-    al_orders = []
-    for order in order_list:
-        as_orders.append(order[0])
-        al_orders.append(order[1])
-    # ren sv starts one order after the first order with as
-    min_as = 1 if min(as_orders) == 0 else min(as_orders)
-    # ren sv starts one order after the first order with al
-    min_al = 1 if min(al_orders) == 0 else min(al_orders)
-    order_as_is_present = False
-    order_al_is_present = False
-    sv_as_present = False
-    sv_al_present = False
-    for order in order_list:
-        if order[0] == min_as + 1:
-            order_as_is_present = True
-            if order[-2] != 0:
-                sv_as_present = True
-        if order[1] == min_al + 1:
-            order_al_is_present = True
-            if order[-2] != 0:
-                sv_al_present = True
-    if not order_as_is_present:
-        sv_as_present = True
-    if not order_al_is_present:
-        sv_al_present = True
-    return sv_as_present, sv_al_present
+    index_to_check = sv_type.value.index
```

Comment:
```suggestion
    svindex = sv_type.value.index
```
Slightly clearer (`sv` is more specific than `to_check`). Even if `sv` is already in the context, `index` is definitely too generic on its own.

Reply:
[6ee6ddd](https://github.com/NNPDF/pineko/pull/75/commits/6ee6ddd3bfc28bd0ec3882c9970e4923a2502205)